src/Compiler/Service/ServiceParsedInputOps.fs

```diff
@@ -1268,6 +1268,7 @@ module ParsedInput =
                 TryGetCompletionContextInPattern false pat (Some(PatternContext.PositionalUnionCaseField(None, id.Range))) pos
             | [ SynPat.Paren (SynPat.Tuple _ | SynPat.Named _ as pat, _) ] ->
                 TryGetCompletionContextInPattern false pat (Some(PatternContext.PositionalUnionCaseField(Some 0, id.Range))) pos
+            | [] when rangeContainsPos id.Range pos -> Some(CompletionContext.Pattern PatternContext.Other)
```

Comment:
Just for my knowledge, why is this change needed?

Reply:
This establishes `CompletionContext.Pattern PatternContext.Other` when the caret is on a long identifier in a pattern. Before, we'd get no completion context and so no filtering would take place.

Reply:
In fact, this should kick in even when arg pats aren't empty. Let me fix it quickly.

Reply:
Thanks!